### PR TITLE
Refactor macOS Getting Started

### LINKED ISSUE #2461: Simplify install instructions on `macOS install` page
Concrete propositions.
1. Move system requirements to the bottom of the screen or to a pop-up - make the blue "get sdk" closer to the top
2. Consider two different audiences - audiences that want hello world to work with no IDE, and audiences that prefer they not use a terminal ever, and make it as easy as possible for those audiences to find what they are looking for - e.g. if you have Intellij and install flutter plugin, how many of the commandline steps will you ever need to do.
3. Load the flutter version number server side rather than clientside in javascript - then the "loading" button will work without having firewall issues and the commandline can be exactly up to date instead of flutter_macos_vX.X.X-stable.zip 

Reason:
* Flutter is reaching a broad base of users. Some are very comfortable with commandline, and some are not.
* Flutter is growing so rapidly that the zero-hour experience continues to be important. (Thanks for having that tag)

awesome job with flutter documentation

### LINKED ISSUE #2529: Include option to get Flutter by cloning repository
Would be nice refer another option to get flutter is cloning the repository.

https://github.com/flutter/website/edit/master/src/docs/get-started/install/_get-sdk.md

### LINKED ISSUE #4995: Clean up 'macOS install' page
Those installation instructions are hard to follow. It's obvious they are written by pros that could not anticipate the problems of intermediate users.

**I would suggest 3 changes.**
1) There are a lot of options (like "if you want to use git …" or "please click this link if you want to add the path permanently [what exactly IS what you want]"). Why not writing a step-by-step-installation-guide which the user just has to follow step by step?

2) There are no concrete examples. Like telling the beginner, hey, we assume your username/-directory is joe and then giving instructions on how exactly to install. Or: We assume your flutter-folder lies in `~/path/to/dev/` , so just type `cd ~/path/to/dev/` to switch to the folder.
3) command ` export PATH="$PATH:`pwd`/flutter/bin"` does not work for me.

### LINKED ISSUE #5374: Add `export` command to `.zprofile` to macOS install page for Mac Catalina or higher
## The problem
I had a problem installing flutter at my Macbook Air 2017 (OS version = Big Sur).

## Step by step to reproduce
- After doing exactly equal as the installing documentation, I restarted the computer
- tried to run `flutter doctor`
- Terminal was logging `command not found ` 

## The Solution
- I googled some solutions and I found that Mac Catalina or Higher requires Full Disk access to Terminal to get Flutter working, but I have no evidence it works (Anyway, I allowed this in preferences).
- I found out that Mac Catalina or higher uses `zsh` with a `.zprofile` (not `.zshrc` nor `.zshenv` etc).
Then I added this to the `.zprofile`: 
`export PATH=$HOME/FLUTTER_DIR/flutter/bin:$PATH`

AND..... IT WORKS! 

## Proposal

Consider adding a note at Get Started Documentation.

Best regards.

### LINKED ISSUE #7475: Getting started instructions for using Xcode to build a macOS app
### What information is missing?

> I couldn't find instructions in https://docs.flutter.dev/get-started for using Xcode to build a macOS app.

I tried:
1. `flutter create my_app`
2. Open `my_app/macos/Runner.xcworkspace` in Xcode.
3. Build.

It failed with:
```
error: Unable to load contents of file list: '[...]/my_app/macos/Flutter/ephemeral/FlutterInputs.xcfilelist' (in target 'Flutter Assemble' from project 'Runner')
error: Unable to load contents of file list: '[...]/my_app/macos/Flutter/ephemeral/FlutterOutputs.xcfilelist' (in target 'Flutter Assemble' from project 'Runner')
```

I got it working by using the `flutter` command-line tool to build the project once before opening the workspace in Xcode. e.g.:
1. `flutter create my_app`
2. `cd my_app`
3. `flutter build macos`
6. Open `my_app/macos/Runner.xcworkspace` in Xcode.
7. Build.

These steps, or the preferred equivalent, should be documented on docs.flutter.dev.

### How would you like us to fix this problem?

Include a section on using Xcode to build a macOS app on https://docs.flutter.dev/get-started.

### LINKED ISSUE #7647: Only .bash_profile should be edited if the Flutter plugin for Android Studio is going to be used
The getting started [instructions](https://flutter.dev/docs/get-started/install/macos#update-your-path) for Mac contain this phrase:
"edit $HOME/.bash_profile or $HOME/.bashrc."

The reference to `.bashrc` may be causing some problems for users of the Flutter plugin for Android Studio. People blame the plugin when cocopods is not found, or when other environment variables are missing. The plugin forks a process to run `flutter` but does not use a shell, so any definitions in `.bashrc` will not be available.

Recent examples:
https://github.com/flutter/flutter-intellij/issues/5727
https://youtrack.jetbrains.com/issue/WEB-52332

### LINKED ISSUE #9520: Fix instructions to install CocoaPods
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Open the instructions: https://docs.flutter.dev/get-started/install/macos#optional-deployment-procedures
2. Run `sudo gem install cocoapods` (not on an M1)
3. See the error (below)

### Actual results

The following error happens.

```
ERROR:  Error installing cocoapods:
	ERROR: Failed to build gem native extension.

    current directory: /Library/Ruby/Gems/2.6.0/gems/ffi-1.16.3/ext/ffi_c
```

Granted, there are other issues reporting difficulties installing CocoaPods with similar looking errors, but they are to do with either running on an M1, or not being able to find the SDK. From what I gather, my issue is not one of those. I'm not running on an M1, and there was no error about not finding the SDK (and forcing the SDKROOT had no effect).

Yes, one could say this is a third party package issue which should be reported to cocoapods or ffi, however I also think that Flutter's "getting started" page should probably try to give instructions that work, so my intention with this issue is not necessarily to ask for help but to improve the getting started instructions.

Since I couldn't get gem to work, I ended up installing cocoapods with homebrew and it worked fine.

### Logs

<details>
<summary>Logs</summary>

```console
$ sudo gem install cocoapods
Password:
Building native extensions. This could take a while...
ERROR:  Error installing cocoapods:
	ERROR: Failed to build gem native extension.

    current directory: /Library/Ruby/Gems/2.6.0/gems/ffi-1.16.3/ext/ffi_c
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby -I /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0 -r ./siteconf20231015-50025-g9ytc1.rb extconf.rb
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/$(RUBY_BASE_NAME)
	--with-ffi_c-dir
	--without-ffi_c-dir
	--with-ffi_c-include
	--without-ffi_c-include=${ffi_c-dir}/include
	--with-ffi_c-lib
	--without-ffi_c-lib=${ffi_c-dir}/lib
	--enable-debug
	--disable-debug
	--enable-system-libffi
	--disable-system-libffi
	--with-libffi-config
	--without-libffi-config
	--with-pkg-config
	--without-pkg-config
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:467:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:546:in `block in try_link0'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/tmpdir.rb:93:in `mktmpdir'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:543:in `try_link0'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:570:in `try_link'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:672:in `try_ldflags'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:1832:in `pkg_config'
	from extconf.rb:9:in `system_libffi_usable?'
	from extconf.rb:46:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /Library/Ruby/Gems/2.6.0/extensions/universal-darwin-22/2.6.0/ffi-1.16.3/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /Library/Ruby/Gems/2.6.0/gems/ffi-1.16.3 for inspection.
Results logged to /Library/Ruby/Gems/2.6.0/extensions/universal-darwin-22/2.6.0/ffi-1.16.3/gem_make.out

$ cat /Library/Ruby/Gems/2.6.0/extensions/universal-darwin-22/2.6.0/ffi-1.16.3/mkmf.log
"pkg-config --exists libffi"
| pkg-config --libs libffi
=> "-lffi\n"
"xcrun clang -o conftest -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin22 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC conftest.c  -L. -L/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib -L. -L/AppleInternal/Library/BuildRoots/d9889869-120b-11ee-b796-7a03568b17ac/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.5.Internal.sdk/usr/local/lib     -lruby.2.6   "
In file included from conftest.c:1:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby.h:33:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/ruby.h:24:10: fatal error: 'ruby/config.h' file not found
#include "ruby/config.h"
         ^~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/ruby.h:24:10: note: did not find header 'config.h' in framework 'ruby' (loaded from '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks')
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */


```

</details>

### Flutter Doctor output

<details>
<summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.7, on macOS 13.5.1 22G90 darwin-x64, locale
    en-AU)
[✗] Android toolchain - develop for Android devices
    ✗ ANDROID_HOME = /Users/ryan/opt/android-sdk
      but Android SDK not found at this location.
[!] Xcode - develop for iOS and macOS (Xcode 15.0)
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] Connected device (2 available)
[✓] Network resources
```

</details>

### PR DESCRIPTION
**Pull request by atsansone**
Total refactor of macOS install.

Fixes #2461
Fixes #2529
Fixes #4995
Fixes #5374
Fixes #7475
Fixes #7647
Fixes #9520

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

**Comment by parlough** [comment]
Is this ready for any further tech or copy review? It's not clear with the `st.WIP` label.

File: src/_includes/docs/install/compiler/android.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+1. Before agreeing to the terms of each license,
+   read each with care.
+
+#### Troubleshooting licensing issues
+{:.no_toc}
+
+<details markdown="1">
+<summary>How to fix the error of finding Java install</summary>
+
+You may have an issue with the Android SDK locating the Java SDK.
```
Comment (sfshaza2) [inline src/_includes/docs/install/compiler/android.md:165]:
```suggestion
You might have an issue with the Android SDK locating the Java SDK.
```

File: src/_includes/docs/install/flutter-doctor.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+   {% assign prompt1='$' %}
+   {% assign prompt2='$' %}
+{% else -%}
+   {% assign path='/usr/bin/flutter' %}
+   {% assign terminal='a shell' %}
+   {% assign prompt1='$' %}
+   {% assign prompt2='$' %}
+{% endcase -%}
+
+### Run Flutter Doctor
```
Comment (sfshaza2) [inline src/_includes/docs/install/flutter-doctor.md:36]:
```suggestion
### Run Flutter doctor
```

File: src/_includes/docs/install/reqs/windows/base.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+{{site.alert.important}}
+Perform this guide in sequence. Skipping steps can cause errors.
```
Comment (sfshaza2) [inline src/_includes/docs/install/reqs/windows/base.md:2]:
```suggestion
  Perform this guide in sequence. Skipping steps can cause errors.
```
